nvfuser/contrib/nn/normalization.py

```diff
@@ -0,0 +1,741 @@
+    if NamedAxis.CHANNEL in stat_axes:
+        stat_dims.append(channel_dim)
+        stat_dims_nobatch.append(channel_dim)
+        num_stats = fd.ops.mul(num_stats, num_channels)
+    x_reduction_axes = [ax for ax in range(num_dims) if ax not in stat_dims]
+    num_features = fd.define_scalar(1)
+    for ax in x_reduction_axes:
+        num_features = fd.ops.mul(num_features, dyn_shape[ax])
+
+    if use_input_stats or running_mean is None:
+        # In NVFuser Python we pass correction=1 to request unbiased variance calculation
+        x_var, x_mean = fd.ops.var_mean(x, x_reduction_axes, int(unbiased))
+        if running_mean is not None:
+            one = fd.define_scalar(1.0)
+            rev_momentum = fd.ops.sub(one, momentum)
+
+            # do running mean with momentum
+            current_mean_hat = fd.ops.mul(x_mean, momentum)
+            mean_hat = fd.ops.mul(running_mean, rev_momentum)
+            new_mean_hat = fd.ops.add(mean_hat, current_mean_hat)
+
+            # If computing stats for each instance, we don't want to keep those
+            # for our running mean calculation, so we sum them here
+            new_mean_sum = (
+                fd.ops.sum(new_mean_hat, [0])
+                if NamedAxis.BATCH in stat_axes
+                else new_mean_hat
+            )
+
+            rev_batch_size = fd.ops.reciprocal(batch_size)
+            new_mean_channels_only = fd.ops.mul(new_mean_sum, rev_batch_size)
+            if x_datatype in [nvfuser.DataType.Half, nvfuser.DataType.BFloat16]:
+                new_mean_channels_only = fd.ops.cast(new_mean_channels_only, x_datatype)
+            fd.add_output(new_mean_channels_only, alias_input=running_mean)
+
+            # running var calculation
+            x_var_unbiased = x_var
+            if not unbiased:
+                # multiply by correction to go from biased to unbiased estimate
+                b2ub = fd.ops.div(
+                    num_features, fd.ops.sub(num_features, fd.define_scalar(1))
+                )
+                x_var_unbiased = fd.ops.mul(x_var, b2ub)
+
+            current_var_hat = fd.ops.mul(x_var_unbiased, momentum)
+            var_hat = fd.ops.mul(running_var, rev_momentum)
+            new_var_hat = fd.ops.add(var_hat, current_var_hat)
+
+            # See above about reducing over batch dim for running stats
+            new_var_sum = (
+                fd.ops.sum(new_var_hat, [0])
+                if NamedAxis.BATCH in stat_axes
+                else new_var_hat
+            )
+
+            new_var_channels_only = fd.ops.mul(new_var_sum, rev_batch_size)
+            if x_datatype in [nvfuser.DataType.Half, nvfuser.DataType.BFloat16]:
+                new_var_channels_only = fd.ops.cast(new_var_channels_only, x_datatype)
+            fd.add_output(new_var_channels_only, alias_input=running_var)
+
+        mean = x_mean
+        mean_bcast = fd.ops.broadcast_in_dim(mean, dyn_shape, stat_dims)
```

Comment:
These could be replaced with `broadcast` calls. We would keep a list of `bool`s that is `True` only for the stat dims like `is_bcast_dim = [d in stat_dims for d in range(num_dims)]` then call `mean_bcast = fd.ops.broadcast(mean, is_bcast_dim)` here. We'd need the same thing but dropping batch dim like `is_bcast_dim_nobatch`.

The difference between `broadcast` and `broadcast_in_dim` is that the latter performs the broadcast I mentioned above followed by an `expand` to the requested size. But we don't need to do an expand in order to use a broadcast tensor in an op. cc @kevinstephano

Reply:
Pushed this change, after which we no longer hit #469. New python definition:
```python
def nvfuser_fusion_id0(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(symbolic_sizes=[-1, -1, -1, -1, -1], contiguity=[True, None, True, True, True], dtype=DataType.Float, is_cpu=False)
    T1 = fd.define_tensor(symbolic_sizes=[-1], contiguity=[None], dtype=DataType.Float, is_cpu=False)
    T2 = fd.define_tensor(symbolic_sizes=[-1], contiguity=[None], dtype=DataType.Float, is_cpu=False)
    S3 = fd.define_scalar(None, dtype=DataType.Double)
    S4 = fd.define_scalar(None, dtype=DataType.Double)
    S5, S6, S7, S8, S9, = fd.ops.tensor_sizes(T0)
    S10 = fd.define_scalar(1, dtype=DataType.Int)
    S11 = fd.ops.mul(S10, S5)
    S12 = fd.ops.mul(S11, S6)
    S13 = fd.define_scalar(1, dtype=DataType.Int)
    S14 = fd.ops.mul(S13, S7)
    S15 = fd.ops.mul(S14, S8)
    S16 = fd.ops.mul(S15, S9)
    T17, T18 = fd.ops.var_mean(T0, axes=[2, 3, 4], correction=0, keepdim=False)
    T19 = fd.ops.broadcast(T18, is_broadcast_dim=[False, False, True, True, True])
    T20 = fd.ops.sub(T0, T19)
    T21 = fd.ops.add(T17, S4)
    T22 = fd.ops.rsqrt(T21)
    T23 = fd.ops.broadcast(T22, is_broadcast_dim=[False, False, True, True, True])
    T24 = fd.ops.mul(T20, T23)
    T25 = fd.ops.broadcast(T1, is_broadcast_dim=[True, False, True, True, True])
    T26 = fd.ops.mul(T24, T25)
    T27 = fd.ops.broadcast(T2, is_broadcast_dim=[True, False, True, True, True])
    T28 = fd.ops.add(T26, T27)
    fd.add_output(T28)
    fd.add_output(T18)
    fd.add_output(T22)
```